cmake/kokkos_arch.cmake

```diff
@@ -358,23 +358,18 @@ ENDIF()
 
 
 IF (KOKKOS_ARCH_KAVERI)
-  SET(KOKKOS_ARCH_ROCM 701)
 ENDIF()
 
 IF (KOKKOS_ARCH_CARRIZO)
-  SET(KOKKOS_ARCH_ROCM 801)
 ENDIF()
 
 IF (KOKKOS_ARCH_FIJI)
-  SET(KOKKOS_ARCH_ROCM 803)
 ENDIF()
 
 IF (KOKKOS_ARCH_VEGA)
-  SET(KOKKOS_ARCH_ROCM 900)
 ENDIF()
 
 IF (KOKKOS_ARCH_GFX901)
-  SET(KOKKOS_ARCH_ROCM 901)
```

Comment:
We need to comment on why we leave empty if-statements here.

Reply:
Do we want to keep the if-statements at all? I couldn't quite convince myself if it's better to remove them or keep them empty.

Removing these architectures completely might break someone's code even if we are not using it at all.

Reply:
I don't understand what's the logic to keep this?

Reply:
The only reason I could come up with is 'A placeholder for future flags, so we haven't "forgotten" about considering these architectures'.

Reply:
I don't think you can run HIP on hardware older than FIJI and I don't think we want to support anything older than Vega

Reply:
So you are proposing that we remove all the architectures where we have empty flags (including `RYZEN`)?

Reply:
ryzen is a cpu. I don't think it has anything to do with rocm. But yeah I would remove these architectures and we can add back what we want later.